### PR TITLE
Test-DbaPowerPlan, fixed a bug and made it more 1.0

### PR DESCRIPTION
**Pull request by ctrlbold**

### CODE REVIEW

File: functions/Test-DbaPowerPlan.ps1 [hunk 1]
```diff
@@ -59,71 +59,116 @@ Test-DbaPowerPlan -ComputerName sqlserver2014a -Detailed
...
+		[switch]$Silent
 	)
-
-	BEGIN
+	
+	begin
 	{
+		If ($Detailed)
+		{
+			Write-Warning "Detailed is deprecated and will be removed in dbatools 1.0"
```
Comment (FriedrichWeinmann) [inline functions/Test-DbaPowerPlan.ps1]: Should be:
`Write-Message -Message "Detailed is deprecated and will be removed in dbatools 1.0" -Level Warning -Once "Deprecated_Parameter_Detail"`

File: functions/Test-DbaPowerPlan.ps1 [hunk 2]
```diff
@@ -59,71 +59,116 @@ Test-DbaPowerPlan -ComputerName sqlserver2014a -Detailed
...
-			try
+			$Server = Resolve-DbaNetworkName -ComputerName $computer -Credential $credential
+			
+			$Computer = $server.ComputerName
+			
+			if (!$Computer)
 			{
-				Write-Verbose "Testing connection to $server and resolving IP address"
-				$ipaddr = (Test-Connection $server -Count 1 -ErrorAction Stop).Ipv4Address | Select-Object -First 1
+				Stop-Function -Message "Couldn't resolve hostname. Skipping." -Continue
```
Comment (FriedrichWeinmann) [inline functions/Test-DbaPowerPlan.ps1]: Add parameter target:
`-Target $Computer`

File: functions/Test-DbaPowerPlan.ps1 [hunk 3]
```diff
@@ -59,71 +59,116 @@ Test-DbaPowerPlan -ComputerName sqlserver2014a -Detailed
...
+			{
+				$powerplans = Get-CimInstance -CimSession $cimsession -classname Win32_PowerPlan -Namespace "root\cimv2\power" -ErrorAction Stop | Select-Object ElementName, InstanceID, IsActive
 			}
 			catch
 			{
-				Write-Warning "Can't connect to WMI on $server"
-				return
+				if ($_.Exception -match "namespace")
+				{
+					Stop-Function -Message "Can't get Power Plan Info for $Computer. Unsupported operating system." -Continue -InnerErrorRecord $_
```
Comment (FriedrichWeinmann) [inline functions/Test-DbaPowerPlan.ps1]: Add parameter target:
`-Target $Computer`

File: functions/Test-DbaPowerPlan.ps1 [hunk 4]
```diff
@@ -59,71 +59,116 @@ Test-DbaPowerPlan -ComputerName sqlserver2014a -Detailed
...
+				}
+				else
+				{
+					Stop-Function -Message "Can't get Power Plan Info for $Computer. Check logs for more details." -Continue -InnerErrorRecord $_
```
Comment (FriedrichWeinmann) [inline functions/Test-DbaPowerPlan.ps1]: Add parameter target:
`-Target $Computer`

File: functions/Test-DbaPowerPlan.ps1 [hunk 5]
```diff
@@ -59,71 +59,116 @@ Test-DbaPowerPlan -ComputerName sqlserver2014a -Detailed
...
+			else
+			{
+				$bpPowerPlan.ElementName = $($powerplans | Where-Object { $_.InstanceID.Split('{')[1].Split('}')[0] -eq $bpPowerPlan.InstanceID }).ElementName
+				if ($null -eq $bpPowerplan.ElementName)
+				{
+					$bpPowerPlan.ElementName = "You do not have the high performance plan installed on this machine."
+				}
+			}
+			
 			Write-Verbose "Recommended GUID is $($bpPowerPlan.InstanceID) and you have $($powerplan.InstanceID)"
```
Comment (FriedrichWeinmann) [inline functions/Test-DbaPowerPlan.ps1]: Should be:
`Write-Message -Level Verbose -Message "Recommended GUID is $($bpPowerPlan.InstanceID) and you have $($powerplan.InstanceID)"`

File: functions/Test-DbaPowerPlan.ps1 (hunk 1 above)
Comment (ctrlbold) [inline functions/Test-DbaPowerPlan.ps1]: Thank you. I removed the once, though, i dont wanna waste some metadata on that it.

**Pull request merged**